mpsl/README.rst

```diff
@@ -1,4 +1,4 @@
-.. _mpsl:
+.. _mpsl_readme:
```

Comment:
Keep using `_mpsl` only to be consistent with other libraries?

Reply:
To align with those libraries that use _readme for files that are not README, but rather integration notes? I think that's strange.

Reply:
The highest level RST files for each library do not have ``_readme`` in their IDs so I would also incline towards leaving just ``_mpsl`` here.
Whether having the ``readme`` suffix for other docs like integration notes is correct - this is another discussion and I agree this could be improved.